### ISSUE TITLE
HashLock returns lots of false positives

### ISSUE THREAD

**Issue by metagon** [metagon/ithildin#5]
This pattern detection strategy needs to be improved to avoid lots of false positives. Due to the nature of the pattern it is often hard to discern between hash lock and accessing data structures, s.a. mappings, where the input gets hashed to get the key. Furthermore this strategy needs to be able to detect hash lock instances where the hash is stored in a storage variable by itself, i.e. not in a mapping.

**Comment by metagon** [comment]
Fixed. Will close the issue once develop is merged into the master branch.

**Comment by metagon** [comment]
Fixed with merging #12
